services/motion/builtin/move_request.go

```diff
@@ -189,6 +225,98 @@ func (mr *moveRequest) deviatedFromPlan(ctx context.Context, waypoints state.Way
 	return state.ExecuteResponse{}, nil
 }
 
+// getTransientDetections returns a list of geometries in their relative position (with respect to the base)
+// and another list of geometries in their absolute positions (with respect to the world) as observed by camName.
+// Relative position geometries are used for constructing a plan, since the base plans in its own frame.
+// Absolute position geometries are used for checking a plan for collisions.
+func (mr *moveRequest) getTransientDetections(
+	ctx context.Context,
+	visSrvc vision.Service,
+	camName resource.Name,
+	currentPosition *referenceframe.PoseInFrame,
+) ([]*referenceframe.GeometriesInFrame, []*referenceframe.GeometriesInFrame, error) {
+	mr.logger.CDebugf(ctx,
+		"proceeding to get detections from vision service: %s with camera: %s",
+		visSrvc.Name().ShortName(),
+		camName.ShortName(),
+	)
+	// any obstacles specified by the worldstate of the moveRequest will also re-detected here.
+	// get detections from vision service
+	detections, err := visSrvc.GetObjectPointClouds(ctx, camName.Name, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+	mr.logger.CDebugf(ctx, "got %d detections", len(detections))
+
+	// determine transform of camera to base
+	cameraOrigin := referenceframe.NewPoseInFrame(camName.ShortName(), spatialmath.NewZeroPose())
+	cameraToBase, err := mr.fsService.TransformPose(ctx, cameraOrigin, mr.kinematicBase.Name().ShortName(), nil)
+	if err != nil {
+		// here we make the assumption the camera is coincident with the world
+		mr.logger.CDebugf(ctx,
+			"we assume the base named: %s is coincident with the camera named: %s due to err: %v",
+			mr.kinematicBase.Name().ShortName(), camName.ShortName(), err.Error(),
+		)
+		cameraToBase = cameraOrigin
+	}
+	mr.logger.CDebugf(ctx, "cameraToBase transform: %v", spatialmath.PoseToProtobuf(cameraToBase.Pose()))
+
+	// detections in the world frame used for CheckPlan
+	absoluteGeoms := []spatialmath.Geometry{}
+	// detection in the base frame used for Replan
+	relativeGeoms := []spatialmath.Geometry{}
```

Comment:
That we need both of these is a red flag for me. Not sure exactly what the solution is but I will look more closely at this and see what we can do.

Reply:
looking that this i think I might have made a mistake. I think all we actually care about is `absoluteGeoms`

Reply:
i think the solution is to only care about absolute geoms as a return value of this function.
However, to do this we then will need to update the existing impl of checkplan to consume obstacles in their relative position rather than their absolute position.

Reply:
ok, my final thought is that we should only care about `relativeGeoms`.
My reasoning is that all static elements of the worldState are already with respect to the base frame.
Ultimately, the current impl of `CheckPlan` would need to change slightly.

Reply:
I will hold off on pushing until we agree on what should be done here.
LMK @biotinker @raybjork

Reply:
we now only return relativeGeoms, so I will mark this as resolved.